### PR TITLE
Data API v2.4.0 Changes

### PR DESCRIPTION
**Pull request by mDuo13**
This updates the dev portal Data API docs with the changes from https://github.com/ripple/rippled-historical-database/pull/183 and adds some more changes (which will hopefully get incorporated into that PR).

Also:

- Updates the Data API v2 tool to add the "Get Validator Manifests" method (added in v2.3.7) and remove the two methods removed in v2.4.0
- Changes the default validator key in the API tool to one that reports recent data
- Fixes a CSS problem that was causing API tool responses to overlap the right column.

This should not be merged until after the Data API PR is merged.

### CODE REVIEW

File: content/references/data-api.md [hunk 1]
```diff
@@ -2675,10 +2675,23 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
...
+| `chain`                 | String                          | Ledger hash chain which this validator is currently following. The value `main` indicates the main network and `altnet` indicates the XRP Test Network. Other forks are named `chain.{NUMBER}`, where `{NUMBER}` is a unique number for each fork. |
+| `unl`                   | Bool                            | True if the validator is part of the ledger chain's recommended UNL. |
+| `current_index`         | Number                          | Ledger index of most recently validated ledger. |
+| `partial`               | Bool                            | True if the most recent validation was a partial one. |
+| `agreement_1h`          | Agreement Object                | Object containing agreement stats for the most recent hour. |
+| `aggreement_24h`        | Agreement Object                | Object containing agreement stats for the most recent 24 hour period. |
```
Comment (jhaaaa) [inline content/references/data-api.md]: tiny typo: change `aggreement_24h` to `agreement_24h`

File: content/references/data-api.md [hunk 2]
```diff
@@ -2693,10 +2706,24 @@ Response:
 ```
 200 OK
 {
-  "domain": "ripple.com",
-  "domain_state": "verified",
-  "last_datetime": "2016-06-07T01:22:59.929Z",
-  "validation_public_key": "n949f75evCHwgyP4fPVgaHqNHxUVN15PsJEZ3B3HnXPcPjcZAoy7",
+  "validation_public_key": "nHBidG3pZK11zQD6kpNDoAhDxH6WLGui6ZxSbUx7LSqLHsgzMPec",
```
Comment (jhaaaa) [inline content/references/data-api.md:2712]: Should we change the `n949f75evCHwgyP4fPVgaHqNHxUVN15PsJEZ3B3HnXPcPjcZAoy7` value in the example request a few lines above to this updated value `nHBidG3pZK11zQD6kpNDoAhDxH6WLGui6ZxSbUx7LSqLHsgzMPec`?

File: content/references/data-api.md [hunk 3]
```diff
@@ -2736,86 +2763,37 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
 | Field                   | Value                           | Description      |
 |:------------------------|:--------------------------------|:-----------------|
 | `result`                | String                          | The value `success` indicates that the body represents a successful response. |
-| `last_datetime`         | Integer                         | Number of links returned. |
-| `validation_public_key` | String - Base-58 [Public Key][] | Validator public key of this validator. |
-
-
-#### Example
+| `count`                 | Integer                         | Number of validators returned. |
+| `validators`            | Array of [Validator Objects][]  | List of validators active in the last 24 hours. |
```
Comment (jhaaaa) [inline content/references/data-api.md:2770]: `[Validator Objects][]` isn't showing up as a link in the generated HTML. Do we want to change this to [Validator Objects](#validator-objects)? Or handle it some other way?

File: content/references/data-api.md [hunk 4]
```diff
@@ -2736,86 +2763,37 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
...
+| `validation_public_key` | String - Base-58 [Public Key][] | This validator's validator public key. |
+| `domain`                | String                          | (May be omitted) The DNS domain associated with this validator. |
+| `chain`                 | String                          | Ledger hash chain which this validator is currently following. The value `main` indicates the main network and `altnet` indicates the XRP Test Network. Other forks are named `chain.{NUMBER}`, where `{NUMBER}` is a unique number for each fork. |
+| `unl`                   | Bool                            | True if the validator is part of the ledger chain's recommended UNL. |
+| `current_index`         | Number                          | Ledger index of most recently validated ledger. |
+| `partial`               | Bool                            | True if the most recent validation was a partial one. |
+| `agreement_1h`          | Agreement Object                | Object containing agreement stats for the most recent hour. |
+| `aggreement_24h`        | Agreement Object                | Object containing agreement stats for the most recent 24 hour period. |
```
Comment (jhaaaa) [inline content/references/data-api.md]: tiny typo: change `aggreement_24h` to `agreement_24h`

File: content/references/data-api.md [hunk 5]
```diff
@@ -2824,48 +2802,40 @@ Response:
...
-    {
-      "count": 27,
-      "first_datetime": "2016-06-08T02:09:17.275Z",
-      "last_datetime": "2016-06-08T02:09:17.383Z",
-      "ledger_hash": "1DC9245CDBFE2C640ACD766DB4AF1DE66F6E92A8EE78F628281A6568760DBAB2",
-      "reporter_public_key": "n9JySgyBVcQKvyDoeRKg7s2Mm6ZcFHk22vUZb3o1HSosWxcj9xPt",
-      "signature": "3045022100E07D8CB801EC7AC98DB1DED813D49AE1FFE3C4CB027314EB6ED1BA7796653DE902204E65E96B0961AC09D8D7542EC59B3CE2ECAE6BC02557A4D1C0385DED00445329",
-      "validation_public_key": "n949f75evCHwgyP4fPVgaHqNHxUVN15PsJEZ3B3HnXPcPjcZAoy7"
-    },
+  "count": 141,
```
Comment (jhaaaa) [inline content/references/data-api.md:2809]: Do we want to provide a sample response in which the `count` matches the number of validator objects in the array? Too much? 🤓

File: content/references/data-api.md [hunk 6]
```diff
@@ -2824,48 +2802,40 @@ Response:
...
 
 
 
-## Get Validations
-[[Source]<br>](https://github.com/ripple/rippled-historical-database/blob/develop/api/routes/network/getValidations.js "Source")
+## Get Validator Manifests
+[[Source]<br>](https://github.com/ripple/rippled-historical-database/blob/develop/api/routes/network/getManifests.js "Source")
 
-Retrieve validation votes, including votes for ledger versions that are outside the main ledger chain. _(New in [v2.2.0][])_
+Retrieve manifests signed by a specified validator, used to designate ephemeral key to sign proposals and validations. _(New in [v2.3.7][])_
```
Comment (jhaaaa) [inline content/references/data-api.md]: The `[v2.3.7][]` link isn't showing up in the generated HTML. Sorry if I'm pointing out stuff that you were already planning to resolve! Just mentioning it to be safe. 🙌

Should "ephemeral key" be plural?

File: content/references/data-api.md [hunk 7]
```diff
@@ -2641,7 +2641,7 @@ Response:
 ## Get Validator
 [[Source]<br>](https://github.com/ripple/rippled-historical-database/blob/develop/api/routes/network/getValidators.js "Source")
 
-Get details of a single validator in the [consensus network](consensus.html). _(New in [v2.2.0][])_
+Get details of a single validator in the [consensus network](consensus.html). _(Updated in [v2.4.0][])_
```
Comment (jhaaaa) [inline content/references/data-api.md:2645]: Do we want to provide a Try It button for this method?

File: content/references/data-api.md [hunk 8]
```diff
@@ -2874,41 +2844,43 @@ Retrieve validation votes, including votes for ledger versions that are outside
 *REST*
 
 ```
-GET /v2/network/validations
+GET /v2/network/validators/{pubkey}/manifests
```
Comment (jhaaaa) [inline content/references/data-api.md:2873]: Do we want to provide a Try It button for this method?

File: content/references/data-api.md [hunk 9]
```diff
@@ -2874,41 +2844,43 @@ Retrieve validation votes, including votes for ledger versions that are outside
...
-| `result`      | String                          | The value `success` indicates that the body represents a successful response. |
-| `count`       | Integer                         | Number of validation votes returned. |
-| `marker`      | String                          | (May be omitted) [Pagination](#pagination) marker. |
-| `validations` | Array of [Validation Objects][] | The requested validation votes. |
+| Field       | Value                         | Description                    |
+|:------------|:------------------------------|:-------------------------------|
+| `result`    | String                        | The value `success` indicates that the body represents a successful response. |
+| `count`     | Integer                       | Number of manifests returned.  |
+| `marker`    | String                        | (May be omitted) [Pagination](#pagination) marker. |
+| `manifests` | Array of [Manifest Objects][] | The requested manifests.       |
```
Comment (jhaaaa) [inline content/references/data-api.md:2903]: The `[Manifest Objects][]` link is not showing up in generated HTML. 

It looks like a Manifest Objects section needs to be created?

File: content/references/data-api.md [hunk 10]
```diff
@@ -2994,19 +2962,19 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
 |:-------------|:-----------------------------------------|:-------------------|
 | `result`     | String                                   | The value `success` indicates that the body represents a successful response. |
 | `count`      | Integer                                  | Number of validators returned. |
```
Comment (jhaaaa) [inline content/references/data-api.md]: I think "Number of validators returned." should be "Number of validator daily reports returned."?

File: content/references/data-api.md [hunk 11]
```diff
@@ -2994,19 +2962,19 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
...
-| `validators` | Array of Single Validator Report Objects | Daily reports of this validator's validation votes. |
+| `reports`    | Array of Single Validator Report Objects | Daily reports of this validator's validation votes. |
 
-Each member in the `validators` array is a Single Validator Report Object with data on that validator's performance on that day. Each has the following fields:
+Each member in the `reports` array is a Single Validator Report Object with data on that validator's performance on that day. Each has the following fields:
```
Comment (jhaaaa) [inline content/references/data-api.md]: Consider adding a `#### Single Validator Report Object` heading above this paragraph, just to match style/structure used for other methods?

File: content/references/data-api.md [hunk 12]
```diff
@@ -3088,19 +3055,17 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
...
-| `other_ledgers`         | Integer                         | Number of other ledger versions this validator voted for. If this number is high, that could indicate that this validator was running non-standard or out-of-date software. |
-| `last_datetime`         | Integer                         | The last reported validation vote signed by this validator. |
-| `domain`                | String                          | (May be omitted) The DNS domain associated with this validator. |
-| `domain_state`          | String                          | The value `verified` indicates that this validator has a [verified domain](run-rippled-as-a-validator.html#domain-verification) controlled by the same operator as the validator. The value `AccountDomainNotFound` indicates that the "Account Domain" part of Domain Verification is not set up properly. The value `RippleTxtNotFound` indicates that the ripple.txt step of Domain Verification is not set up properly. |
+Each member in the `reports` array is a Single Validator Report Object with data on that validator's performance on that day. Each has the following fields:
```
Comment (jhaaaa) [inline content/references/data-api.md]: Consider adding a `#### Daily Validator Report Object` heading above this paragraph, just to match style/structure used for other methods?

Should "Single Validator Report Object" be "Daily Validator Report Object"?

File: content/references/data-api.md [hunk 13]
```diff
@@ -3088,19 +3055,17 @@ A successful response uses the HTTP code **200 OK** and has a JSON body with the
 
 Daily Validator Report Object fields:
 
-| Field                   | Value                           | Description      |
```
Comment (jhaaaa) [inline content/references/data-api.md:3091]: SUPER nit about the intro sentence above. Considering changing "Get a validation vote stats" to "Get validation vote stats"?

File: content/references/data-api.md (hunk 5 above)
Comment (mDuo13) [reply content/references/data-api.md:2809]: Too much. We can provide a second object and an ellipsis to indicate that the example is incomplete.

File: content/references/data-api.md (hunk 6 above)
Comment (mDuo13) [reply content/references/data-api.md]: Nah, good catch. And changed it to "an ephemeral key" (one at a time)

File: content/references/data-api.md (hunk 7 above)
Comment (mDuo13) [reply content/references/data-api.md:2645]: Now that the tool is updated, yes.

File: content/references/data-api.md (hunk 9 above)
Comment (mDuo13) [reply content/references/data-api.md:2903]: Good catch. Working on it.

**Comment by mDuo13** [comment]
!rebuild

**Comment by mDuo13** [comment]
(Now that the corresponding changes have been pushed to the Data API repo, this should be good to merge... assuming no more broken links)

**Review by jhaaaa — APPROVED** [review]
LGTM - thank you for this! 🙌

**Pull request merged**